driver-core/src/main/java/com/datastax/driver/core/RequestHandler.java

```diff
@@ -224,6 +225,16 @@ private void setFinalResult(
                 response.getCustomPayload());
       }
       callback.onSet(connection, response, info, statement, System.nanoTime() - startTime);
+      // if the response from the server has warnings, they'll be set on the ExecutionInfo. Log them
+      // here, if enabled.
+      if (logger.isWarnEnabled()
+          && response.warnings != null
+          && !response.warnings.isEmpty()
+          && Boolean.getBoolean(RequestHandler.LOG_REQUEST_WARNINGS_PROPERTY)) {
+        for (String warning : response.warnings) {
+          logger.warn(warning);
```

Comment:
Should we provide a bit more context? Something like "The server generated a warning for a query: {}"... Maybe also include the query string, or a substring if it's too long?